fbgemm_gpu/codegen/training/backward/embedding_backward_split_kernel_warp_template.cu

```diff
@@ -462,7 +462,7 @@ template <
     int32_t kThreadGroupSize,
     bool kUseVecBlocking,
     int32_t embedding_dim,
-    int32_t weight_decay_mode>
+    int32_t weight_decay_mode_v>
```

Comment:
Is there any reason to add `_v` suffix (I assume there is, just for some documentation)

Reply:
`weight_decay_mode` argument will be generated for some implementations in the following line:
```C++
{{ args.split_kernel_args | replace_pta_namespace() | join(",\n ") }}
```
And it's going to conflict with template argument name so I've decided to rename it STL-type_traits-kind-of convention. 

Another interesting question is how branch divergence will affect the performance in new kernel. Maybe we don't need this (and some other) template arguments at all.